mosqagent.h

```diff
@@ -14,6 +14,66 @@
 struct mosqagent_idle_list;
 struct mosqagent_config;
 
+/**
+ * \brief A memory object that stores ownership information.
+ *
+ * Never manipulate the values directly! Always use mqtta_mo_move and
+ * mqtta_mo_set to change pointers. Ownership information is managed
+ * automatically by these functions.
+ */
+struct mqtta_memory_object {
+    void *ptr;
+    bool ownership;
+};
+
```

Comment:
I thought about this: If I hide the struct, only pointers can be used. Which means there are heap objects of structs with meta-information about heap objects (and again the question who has to clean those up). 

My idea was that this struct is embedded directly. The reason for using pointers in the API is that copies of this struct should not be necessary and the user can decide where the struct resides.

Reply:
You are right, and you use it like this in mqtt-clock.c here:

```
    struct mqtta_memory_object mo;
    mqtta_mo_ptr(&mo);
```

I overlooked this part. If you use it like this, you can not hide the struct.

Maybe this is some question of taste. I saw other libraries where you request some kind of handle and the allocation of the struct behind happens dynamically inside the lib.

Reply:
Yes… I am not a fan of handle-based systems, though.This would mean replacing one handle (pointer) by a far less powerful handle just so that the user cannot manipulate one value.

On the other hand: In those cases where the users hand over a pointer, they need to specify the da-allocator anyways. So even with a handle-based system there is so much margin for error that I do not think it is worth the effort.

After all, I wanted to make (my) life a bit easier.

Reply:
There has been much discussion about this topic regarding smart pointers in C++ standardization.

Reply:
Maybe I was just irritated by this comment:
```
/**
 * \brief A memory object that stores ownership information.
 *
 * Never manipulate the values directly! Always use mqtta_mo_move and
 * mqtta_mo_set to change pointers. Ownership information is managed
 * automatically by these functions.
 */
```
So if this library exposes the memory object, I'm fine with that, as long as the intended use is documented.